test/src/unit-conversions.cpp

```diff
@@ -1537,6 +1537,31 @@ NLOHMANN_JSON_SERIALIZE_ENUM(TaskState,
     {TS_COMPLETED, "completed"},
 })
 
+namespace
+{
+// Helper struct to test whether compile error does not trigger upon
+// conversion of an enum in the presence of non-member operator== for
+// user-defined type with "non default" from_json function (#1647).
+struct NonDefaultFromJsonStruct { };
```

Comment:
Could you please move this to the regression tests?

Reply:
Sure thing, I'll try and get it done one of the next few days.